src/async_impl/client.rs

```diff
@@ -150,6 +150,7 @@ struct Config {
     quic_send_window: Option<u64>,
     dns_overrides: HashMap<String, Vec<SocketAddr>>,
     dns_resolver: Option<Arc<dyn Resolve>>,
+    base_url: Option<String>,
```

Comment:
An `Opiton<Url>` might be better?

Reply:
This change should be made in `ClientRef`, not in `Client`